src/main/java/digital/slovensko/autogram/model/ProtectedDSSDocument.java

```diff
@@ -0,0 +1,12 @@
+package digital.slovensko.autogram.model;
+
+import eu.europa.esig.dss.model.DSSDocument;
+
+public interface ProtectedDSSDocument extends DSSDocument {
+    char[] getPassword();
```

Comment:
A fakt radšej rozlišujme ten `openDocumentPassowrd` a `masterPassword` (tak nejak som to našiel na nete pomenované). Prvé je na vôbec zobrazenie dokumentu a druhé je na rôzne permissions - medzi nimi aj editáciu - avšak na podpis nemusí byť heslo potrebné, záleží od nastavenia permissions.

Celé je to o level zložitejšie než sa zdalo na začiatku.

Ak existuje `openDocumentPassowrd`, potrebujeme ho, lebo bez neho nevieme pdf ani zobraziť.

Ak existuje `masterPassword`, nemusí to hneď znamenať, že sa nedá podpisovať.

V DSS dokumentácii ale píšu, že to podpisovanie password protected PDF nejako funguje v takom lite režime. https://ec.europa.eu/digital-building-blocks/DSS/webapp-demo/doc/dss-documentation.html#_password_protected_documents ale do parametrov to berie iba nejaké jedno heslo, takže to bude zrejme iba ten `openDocumentPassword`.

Potom inde v dokumentácii píšu o tom, ako fungujú permissions v PDF: https://ec.europa.eu/digital-building-blocks/DSS/webapp-demo/doc/dss-documentation.html#ProtectedPDF Ak správne rozumiem, tieto "dictionaries" by mali byť uzamykateľné nejakým `masterPassword`, ale napr. taký DocMDP funguje tak, že v podstate podpisom ho zamkneš - resp. ak by ho niekto neskôr porušil, tak tvoj podpis začne byť nevalidný.

Reply:
V prípade `openDocumentPassword` potrebujeme toto heslo avšak použiť na úplne všetko, aj na podpis (tipujem že je to tým, že je PDF enkryptované). Teda vo funkcií SigningJob#signDocumentAsPAdeS nakoniec tak či tak skončíme s tým, že sa bude používať bude masterPassword alebo openDocumentPassword

Reply:
cc @celuchmarek